.github/workflows/c-fortran-test-style.yml

```diff
@@ -19,11 +19,14 @@ jobs:
     - name: Environment setup
       uses: actions/checkout@v3
     - name: Install clang-format
-      run: sudo apt update && sudo apt install clang-format-15
+      run: |
+          wget -O- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          sudo add-apt-repository 'deb http://apt.llvm.org/bullseye/ llvm-toolchain-bullseye-16 main'
+          sudo apt update && sudo apt install clang-format-16
```

Comment:
ah, this wouldn't be needed (I think) if GitHub Actions was using Ubuntu 23.04, but that doesn't seem like it will happen any time soon, and I don't think 20.04 will get `clang-format-16` any time soon. Long story short - I think this is probably our best bet for now.

It looks like `clang-format-16` will be the default on Ubuntu 23.10, and this is a small change, so I figure we can merge now?

Reply:
Oh, but `clang-16` isn't conveniently available on Arch linux yet

Reply:
> It looks like clang-format-16 will be the default on Ubuntu 23.10, and this is a small change, so I figure we can merge now?

Nice, I agree with this